src/services/productService.ts

```diff
@@ -0,0 +1,105 @@
+// Import product model
+import { Product } from "../models/ProductModel";
+
+// Import rating model
+import { Rating } from "../models/RatingModel";
+
+// Import Sequelize and Op from sequelize module
+import { Sequelize, Op } from "sequelize";
+
+export class productService {
+  // This method to add discount information to product information
+  static addDiscountInfo(productInfo: any) {
+    if (productInfo.discount_percentage === 0) {
+      productInfo.price_after_discount = null;
+      return;
+    }
+    const discountValue =
+      productInfo.price * (productInfo.discount_percentage / 100);
+    productInfo.price_after_discount = productInfo.price - discountValue;
+    productInfo.price_after_discount = parseFloat(
+      productInfo.price_after_discount.toFixed(2)
+    );
+  }
+  // This method to get new arrivals products
+  static async getNewArrivalsProducts() {
+    const currentMonth: number = new Date().getMonth() + 1;
+    let targetMonth: number = 0;
```

Comment:
move contant numbers to a general contant file and import it at the places where needed